coding-agent/src/coding_agent/ui/http_server.py

```
@@ -1276,6 +1322,143 @@ async def cancel_session_turn(
+    except KeyError as exc:
+        raise HTTPException(status_code=404, detail=str(exc)) from exc
+    return _workspace_summary_response(entry)
+
+
+@app.delete("/workspaces/{workspace_id}", response_model=WorkspaceCleanupResponse)
+@limiter.limit(RateLimits.CLOSE_SESSION)
+async def cleanup_workspace(
+    request: Request,
+    response: Response,
+    workspace_id: str,
+    auth_context: AuthContext | None = Depends(auth_context_from_headers),
```

Ruff (0.15.12)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)